src/algebra/associated.lean

```diff
@@ -286,64 +294,66 @@ multiset.induction_on s (by simp [mt is_unit_iff_dvd_one.2 hp.not_unit])
     cases hp.div_or_div hps with h h,
     { use [a, by simp],
       cases h with u hu,
-      cases ((irreducible_of_prime (hs a (multiset.mem_cons.2
-        (or.inl rfl)))).is_unit_or_is_unit hu).resolve_left hp.not_unit with v hv,
+      cases (((hs a (multiset.mem_cons.2 (or.inl rfl))).irreducible)
+        .is_unit_or_is_unit hu).resolve_left hp.not_unit with v hv,
       exact ⟨v, by simp [hu, hv]⟩ },
     { rcases ih (λ r hr, hs _ (multiset.mem_cons.2 (or.inr hr))) h with ⟨q, hq₁, hq₂⟩,
       exact ⟨q, multiset.mem_cons.2 (or.inr hq₁), hq₂⟩ }
   end)
 
-lemma dvd_iff_dvd_of_rel_left [monoid α] {a b c : α} (h : a ~ᵤ b) : a ∣ c ↔ b ∣ c :=
+lemma associated.dvd_iff_dvd_left [monoid α] {a b c : α} (h : a ~ᵤ b) : a ∣ c ↔ b ∣ c :=
 let ⟨u, hu⟩ := h in hu ▸ units.mul_right_dvd.symm
 
-lemma dvd_iff_dvd_of_rel_right [monoid α] {a b c : α} (h : b ~ᵤ c) : a ∣ b ↔ a ∣ c :=
+lemma associated.dvd_iff_dvd_right [monoid α] {a b c : α} (h : b ~ᵤ c) : a ∣ b ↔ a ∣ c :=
 let ⟨u, hu⟩ := h in hu ▸ units.dvd_mul_right.symm
 
-lemma eq_zero_iff_of_associated [monoid_with_zero α] {a b : α} (h : a ~ᵤ b) : a = 0 ↔ b = 0 :=
+lemma associated.eq_zero_iff [monoid_with_zero α] {a b : α} (h : a ~ᵤ b) : a = 0 ↔ b = 0 :=
 ⟨λ ha, let ⟨u, hu⟩ := h in by simp [hu.symm, ha],
   λ hb, let ⟨u, hu⟩ := h.symm in by simp [hu.symm, hb]⟩
 
-lemma ne_zero_iff_of_associated [monoid_with_zero α] {a b : α} (h : a ~ᵤ b) : a ≠ 0 ↔ b ≠ 0 :=
-not_congr $ eq_zero_iff_of_associated h
+lemma associated.ne_zero_iff [monoid_with_zero α] {a b : α} (h : a ~ᵤ b) : a ≠ 0 ↔ b ≠ 0 :=
+not_congr h.eq_zero_iff
 
-lemma prime_of_associated [comm_monoid_with_zero α] {p q : α} (h : p ~ᵤ q) (hp : prime p) :
+protected lemma associated.prime [comm_monoid_with_zero α] {p q : α} (h : p ~ᵤ q) (hp : prime p) :
   prime q :=
-⟨(ne_zero_iff_of_associated h).1 hp.ne_zero,
+⟨h.ne_zero_iff.1 hp.ne_zero,
   let ⟨u, hu⟩ := h in
     ⟨λ ⟨v, hv⟩, hp.not_unit ⟨v * u⁻¹, by simp [hv, hu.symm]⟩,
       hu ▸ by { simp [units.mul_right_dvd], intros a b, exact hp.div_or_div }⟩⟩
 
 lemma associated_of_irreducible_of_dvd [cancel_monoid_with_zero α] {p q : α}
```

Comment:
Don't know if this one is worth the dot.
```suggestion
lemma irreducible.associated_of_dvd [cancel_monoid_with_zero α] {p q : α}
```
or, with a shuffling of the arguments,
```suggestion
lemma dvd.associated_of_irreducible [cancel_monoid_with_zero α] {p q : α}
```

Reply:
I'm going for the first option, the `irreducible` namespace, since there are other cases where division implies the symmetric case as well: units, prime elements (see the next declaration!), ...

Reply:
These don't seem to be used anywhere, by the way...